### DISCUSSION TITLE
use fail2ban cause ssh exception

### DISCUSSION THREAD

**Discussion by yulongshi0816** [fail2ban/fail2ban#3709] [Q&A]
when disabling ip in fail2ban ，it will cause other ip cant login with ssh about 1 min please helpme

**Comment by sebres** [comment]
> when disabling ip in fail2ban 

You mean some IP gets banned?

A ban means the incoming connection for IP (or IP:port combination) will be rejected.
That all depends on banning action, but usually it means the incoming connect **from IP**, not **to IP**.
Thus if it mistakenly your host IP (unexpected but), it'd be anyway still possible to connect.
So it is not clear at all how banning of some single IP could cause totale block on ssh port.

Please provide the excerpt of:
`fail2ban-client -d | grep sshd`
as well as the output of `iptables` (`ipset`) or `nft` depending on your banning action:
```
iptables -nL
ipset list
nft list tables
```

> it will cause other ip cant login with ssh about 1 min please helpme

Nobody would be able to help you with so few info.

The issue may be: if the IP is some proxying IP (host IP if it is a container, etc), it is not correct to ban it (instead one needs to find out why it is logged and either fix it or find another log/journal with correct IPs).
If the filter mistakenly detect wrong IP for some reason, please provide the log-excerpt, with the messages caused ban (and your fail2ban version).
Also note the question 3 in [FAQ in our wiki](https://github.com/fail2ban/fail2ban/wiki/How-fail2ban-works).

**Comment by yulongshi0816** [comment]
I'm sorry for providing very little information. The issue I encountered is that I'm using version 1.02 of fail2ban. When I intentionally input the wrong password while using an IP to access 10.11.12.13, that IP gets banned, which is expected. However, when attempting to SSH connect to 10.11.12.13 with another IP after banning that first IP, even with the correct password, the connection is refused. It returns to normal after about a minute.

**Reply by sebres** [reply]
OMG... basically, you said not more than in your initial question...
Something like that was already clear to me (however is it not normal and unexpected). 
And I already asked about more info. Without that it is not possible to help here, sorry.

**Comment by yulongshi0816** [comment]
Untitled.pdf
sorry for little information， this is the command fail2ban-client -d ｜grep sshd。 the ipset
 List and nft list table is empty  the command iptable -nl the last Chain f2b-SSH also empty

**Reply by sebres** [reply]
PDF? Really? It is moveton in IT-world (don't you know, PDF could also contain a virus?)...
Please next time provide the info directly enclosed in \`\`\` (after new-line), see [GH markdown docu](https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax#quoting-code)
Or just attach a txt-file.

But OK, there are enough pdftotext online-tools, so I got this (re-formatted pretty):
<details><summary>fail2ban-client -d ｜grep sshd</summary>

```
['add', 'sshd', 'auto']
['set', 'sshd', 'usedns', 'warn']
['set', 'sshd', 'prefregex', '^<F-MLFID>(?:\\[\\])?\\s*(?:<[^.]+\\.[^.]+>\\s+)?(?:\\S+\\s+)?(?:kernel:\\s?\\ [ *\\d+\\.\\d+\\]:?\\s+)?(?:@vserver_\\S+\\s+)?(?:(?:(?:\\[\\d+\\])?:\\s+[\\[\\(]?sshd(?:\\(\\S+\\))?[\\]\\)]?:?| [\\[\\(]?sshd(?:\\(\\S+\\))?[\\]\\)]?:?(?:\\[\\d+\\])?:?)\\s+)?(?:\\[ID \\d+ \\S+\\]\\s+)?</F-MLFID>(?:(?:error| fatal): (?:PAM: )?)?<F-CONTENT>.+</F-CONTENT>$']
['set', 'sshd', 'maxlines', 1]
['multi-set', 'sshd', 'addfailregex', ['^[aA]uthentication (?:failure|error|failed) for <F-USER>.*?</FUSER> (?:from )?<HOST>( via \\S+)?(?: (?:port \\d+|on \\S+|\\[preauth\\])){0,3}\\s*$', '^User not known to the underlying authentication module for <F-USER>.*?</F-USER> (?:from )?<HOST>(?: (?:port \\d+|on \\S+|\\[preauth\\])){0,3}\\s*$', '^Failed publickey for invalid user <F-USER>(? P<cond_user>\\S+)|(?:(?! from ).)*?</F-USER> from <HOST>(?: (?:port \\d+|on \\S+)){0,2}(?: ssh\ \d*)?(?(cond_user): |(?:(?:(?! from ).)*)$)', '^Failed (?:<F-NOFAIL>publickey</F-NOFAIL>|\\S+) for (? P<cond_inv>invalid user )?<F-USER>(?P<cond_user>\\S+)|(?(cond_inv)(?:(?! from ).)*?|[^:]+)</FUSER> from <HOST>(?: (?:port \\d+|on \\S+)){0,2}(?: ssh\\d*)?(?(cond_user): |(?:(?:(?! from ).)*)$)', '^<F-USER>ROOT</F-USER> LOGIN REFUSED FROM <HOST>', '^[iI](?:llegal|nvalid) user <FUSER>.*?</F-USER> (?:from )?<HOST>(?: (?:port \\d+|on \\S+|\\[preauth\\])){0,3}\\s*$', '^User <FUSER>\\S+|.*?</F-USER> (?:from )?<HOST> not allowed because not listed in AllowUsers(?: (?:port \\d+|on \\S+|\\[preauth\\])){0,3}\\s*$', '^User <F-USER>\\S+|.*?</F-USER> (?:from )?<HOST> not allowed because listed in DenyUsers(?: (?:port \\d+|on \\S+|\\[preauth\\])){0,3}\\s*$', '^User <FUSER>\\S+|.*?</F-USER> (?:from )?<HOST> not allowed because not in any group(?: (?:port \\d+| on \\S+|\\[preauth\\])){0,3}\\s*$', '^refused connect from \\S+ \\(<HOST>\\)', '^Received <FMLFFORGET>disconnect</F-MLFFORGET> from <HOST>(?: (?:port \\d+|on \\S+)){0,2}:\\s*3: .*: Auth fail(?: (?:port \\d+|on \\S+|\\[preauth\\])){0,3}\\s*$', '^User <F-USER>\\S+|.*?</F-USER> (?:from )?<HOST> not allowed because a group is listed in DenyGroups(?: (?:port \\d+|on \\S+|\\[preauth\\])) {0,3}\\s*$', "^User <F-USER>\\S+|.*?</F-USER> (?:from )?<HOST> not allowed because none of user's groups are listed in AllowGroups(?: (?:port \\d+|on \\S+|\\[preauth\\])){0,3}\\s*$", '^<FNOFAIL>pam_[a-z]+\\(sshd:auth\\):\\s+authentication failure;</F-NOFAIL>(?:\\s+(?:(?:logname|e?uid| tty)=\\S*)){0,4}\\s+ruser=<F-ALT_USER>\\S*</F-ALT_USER>\\s+rhost=<HOST>(?:\\s+user=<FUSER>\\S*</F-USER>)?(?: (?:port \\d+|on \\S+|\\[preauth\\])){0,3}\\s*$', '^maximum authentication attempts exceeded for (?:invalid user )?<F-USER>.*?</F-USER> (?:from )?<HOST>(?: (?:port \\d+| on \\S+)){0,2}(?: ssh\\d*)?(?: (?:port \\d+|on \\S+|\\[preauth\\])){0,3}\\s*$', '^User <F-USER>\\S+|.*?</ F-USER> not allowed because account is locked(?: (?:port \\d+|on \\S+|\\[preauth\\])){0,3}\\s*', '^<FMLFFORGET>Disconnecting</F-MLFFORGET>(?: from)?(?: (?:invalid|authenticating)) user <FUSER>\\S+</F-USER> <HOST>(?: (?:port \\d+|on \\S+)){0,2}:\\s*Change of username or service not allowed:\\s*.*\\[preauth\\]\\s*$', '^Disconnecting: Too many authentication failures(?: for <F-USER>\ \S+|.*?</F-USER>)?(?: (?:port \\d+|on \\S+|\\[preauth\\])){0,3}\\s*$', '^<F-NOFAIL>Received <FMLFFORGET>disconnect</F-MLFFORGET></F-NOFAIL> from <HOST>(?: (?:port \\d+|on \\S+)) {0,2}:\\s*11:', '^<F-NOFAIL><F-MLFFORGET>(?:Connection (?:closed|reset)|Disconnect(?:ed|ing))</ F-MLFFORGET></F-NOFAIL>(?: (?:by|from))?(?: (?:invalid|authenticating) user <F-USER>\\S+|.*?</ F-USER>)?(?: from)? <ADDR>(?: (?:port \\d+|on \\S+)){0,2}(?:: (?!Too many authentication failures)[^\ \[]+)?(?: \\[preauth\\])?\\s*$', '^<F-MLFFORGET><F-MLFGAINED>Accepted \\w+</FMLFGAINED></F-MLFFORGET> for <F-USER>\\S+</F-USER> from <HOST>(?:\\s|$)', '^<FNOFAIL>Connection from</F-NOFAIL> <HOST>']]
['set', 'sshd', 'datepattern', '{^LN-BEG}']
['set', 'sshd', 'addjournalmatch', '_SYSTEMD_UNIT=sshd.service', '+', '_COMM=sshd']
['set', 'sshd', 'maxretry', 3]
['set', 'sshd', 'maxmatches', 3]
['set', 'sshd', 'findtime', '10m']
['set', 'sshd', 'bantime', '300']
['set', 'sshd', 'ignorecommand', '']
['set', 'sshd', 'logencoding', 'auto']
['set', 'sshd', 'addlogpath', '/var/log/secure', 'head']
['set', 'sshd', 'addaction', 'iptables-SSH']
['multi-set', 'sshd', 'action', 'iptables-SSH', [
  ['actionstart', "
      { <iptables> -C f2b-SSH -j RETURN >/ dev/null 2>&1; } || { <iptables> -N f2b-SSH || true; <iptables> -A f2b-SSH -j RETURN; }
      for proto in $(echo 'tcp' | sed 's/,/ /g'); do
      { <iptables> -C INPUT -p $proto --dport ssh -j f2b-SSH >/dev/null 2>&1; } || { <iptables> -I INPUT -p $proto --dport ssh -j f2b-SSH; }
      done
  "],
  ['actionstop', "
      for proto in $(echo 'tcp' | sed 's/,/ /g'); do
      <iptables> -D INPUT -p $proto --dport ssh -j f2bSSH
      done
      <iptables> -F f2b-SSH
      <iptables> -X f2b-SSH
  "],
  ['actionflush', '<iptables> -F f2bSSH'],
  ['actioncheck', "
      for proto in $(echo 'tcp' | sed 's/,/ /g'); do
      <iptables> -C INPUT -p $proto -dport ssh -j f2b-SSH
      done
  "],
  ['actionban', '<iptables> -I f2b-SSH 1 -s <ip> -j <blocktype>'],
  ['actionunban', '<iptables> -D f2b-SSH -s <ip> -j <blocktype>'],
  ['name', 'SSH'], ['port', 'ssh'], ['protocol', 'tcp'], 
  ['actname', 'iptables-SSH'],
  ['chain', 'INPUT'], 
  ['blocktype', 'REJECT --reject-with icmp-port-unreachable'], 
  ['returntype', 'RETURN'], 
  ['lockingopt', '-w'],
  ['iptables', 'iptables <lockingopt>'], 
  ['blocktype?family=inet6', 'REJECT --reject-with icmp6-port-unreachable'],
  ['iptables?family=inet6', 'ip6tables <lockingopt>']
]]
['start', 'sshd']
```
</details>

Action `iptables-SSH` doesn't tell me something (we don't have such action at all). 
But at first glance it looks like some derivate of `iptables[type=multiport]`...
However I have no time to check it completely or find out what may be wrong here.
You can try to rewrite your sshd-jail like this:
```ini
[sshd]
banaction = iptables[type=multiport]
# if default action is overwritten (somewhere in default of *.local files):
action = %(action_)s
enabled = true
```
then restart fail2ban (or sshd jail) and try it again.

> Chain f2b-SSH also empty

Strange. It must have at least one rule with RETURN (also related to `actionstart` from your dump it shall contain that).
(And the IPs are not there if no one is banned currently).

If you has nft, you can also try to use `banaction = nftables[type=multiport]` for jails or default section (it is anyway better than pure iptables action without ipset).
